src/toast/ops/sim_gaindrifts.py

```diff
@@ -0,0 +1,322 @@
+            focalplane = ob.telescope.focalplane
+            size = ob.detdata[self.det_data][dets[0]].size
+            fsampl = focalplane.sample_rate.to_value(u.Hz)
+
+            if self.drift_mode == "linear_drift":
+                key1 = (
+                    self.realization * 4294967296 + telescope * 65536 + self.component
+                )
+                counter1 = 0
+                counter2 = 0
+
+                for det in dets:
+                    detindx = focalplane[det]["uid"]
+                    key2 = obsindx * 4294967296 + detindx
+
+                    rngdata = rng.random(
+                        1,
+                        sampler="gaussian",
+                        key=(key1, key2),
+                        counter=(counter1, counter2),
+                    )
+                    gf = 1 + rngdata[0] * self.sigma_drift
+                    gain = (gf - 1) * np.linspace(0, 1, size) + 1
+
+                    ob.detdata[self.det_data][det] *= gain
+
+            elif self.drift_mode == "thermal_drift":
+                fmin = fsampl / (4 * size)
+                # the factor of 4x the length of the sample vector  is
+                # to avoid circular correlations
+                freq = np.logspace(np.log10(fmin), np.log10(fsampl / 2.0), 1000)
+                psd = self.get_psd(freq)
+                det_group = np.unique(focalplane.detector_data[self.focalplane_group])
+                thermal_fluct = []
+                for iw, w in enumerate(det_group):
+                    # simulate a noise-like timestream
+                    gain = sim_noise_timestream(
+                        realization=self.realization,
+                        telescope=ob.telescope.uid,
+                        component=self.component,
+                        obsindx=ob.uid,
+                        # we generate the same timestream for the
+                        # detectors in the same group
+                        detindx=iw,
+                        rate=fsampl,
+                        firstsamp=ob.local_index_offset,
+                        samples=ob.n_local_samples,
+                        freq=freq,
+                        psd=psd,
+                        py=False,
+                    )
+                    thermal_fluct.append(gain)
+                thermal_fluct = np.array(thermal_fluct)
+
+                for det in dets:
+                    detindx = focalplane[det]["uid"]
+                    # we inject a detector mismatch in the thermal thermal_fluctuation
+                    # only if the mismatch !=0
+                    if self.detector_mismatch != 0:
+                        key1 = (
+                            self.realization * 429496123345
+                            + telescope * 6512345
+                            + self.component
+                        )
+                        counter1 = 0
+                        counter2 = 0
+                        key2 = obsindx * 12345676 + detindx
```

Comment:
The starting state of the RNG is determined from `key1`, `key2`, and `counter1`.  Then each sample increments `counter2`.  Since `detindx` and `obsindx` might be a very large 64bit integer, perhaps we could do this:
```
key1 --> (keep the same- based on realization, telescope, component)
key2 --> obsindx
counter1 --> detindx
counter2 --> 0
```

Reply:
Sounds good ! thanks